salt/utils/network.py

```diff
@@ -1958,3 +1959,33 @@ def parse_host_port(host_port):
             raise ValueError('bad hostname: "{}"'.format(host))
 
     return host, port
+
+
+@jinja_filter("filter_by_networks")
+def filter_by_networks(values, networks=None):
+    """
+    Returns a the list of IPs filtered by the optional network list.
+    If no filter is specified all all items are returned.
```

Comment:
Maybe to fulfill @darix user case and the current version of the code (where an `is not None` make it explicit), we need to document that for None all items will be returned, and for an empty list of networks, no match is found.

I still find this weird, but I guess that is OK.

Reply:
I've updated the docstring according to what the code does at the moment. Let's discuss the behavior of `None` vs `[]` in the other thread.